tests/testTransform.py

```diff
@@ -0,0 +1,171 @@
+
+    def testNullTransform(self):
+        """The NullTransform passes through nothing"""
+        inCat = self._generateCatalog()
+        outCat = self._performTransform(measBase.NullTransform, inCat)
+        self.assertEqual(len(inCat), len(outCat))
+        self.assertEqual(outCat.schema.getFieldCount(), 0)
+
+    def testPassThroughTransform(self):
+        """The PassThroughTransform passes through all fields starting with the plugin name"""
+        inCat = self._generateCatalog()
+        outCat = self._performTransform(measBase.PassThroughTransform, inCat)
+        self.assertEqual(len(inCat), len(outCat))
+        self.assertEqual(len(inCat.schema.extract(self.pluginName + "*")), outCat.schema.getFieldCount())
+
+    def testTransformWrapper(self):
+        """The TransformWrapper lets us treat a C++ transform as if it were Python"""
+        inCat = self._generateCatalog()
+        transform = measBase.TransformWrapper(testLib.SillyTransform)
+        outCat = self._performTransform(transform, inCat)
+        self._checkSillyOutputs(inCat, outCat)
+
+    def testApplyCppTransform(self):
+        """Test that we can apply a simple C++ transform"""
+        inCat = self._generateCatalog()
+        sillyControl = testLib.SillyCentroidControl()
+        mapper = afwTable.SchemaMapper(inCat.schema)
+        sillyTransform = testLib.SillyTransform(sillyControl, self.pluginName, mapper)
+        outCat = afwTable.BaseCatalog(mapper.getOutputSchema())
+        outCat.extend(inCat, mapper=mapper)
+        self.assertEqual(len(inCat), len(outCat))
+        sillyTransform(inCat, outCat, makeWcs(), afwImage.Calib())
+        self._checkSillyOutputs(inCat, outCat)
+
+class AlgorithmConfigurationTestCase(utilsTests.TestCase):
+    def testDefaultTransform(self):
+        """By default, we perform no transformations"""
+        self.assertEqual(measBase.BasePlugin.getTransformClass(), measBase.NullTransform)
+
+    def testWrapAlgorithm(self):
+        """Test that the appropriate transform is provided for wrapped algorithms"""
+        # By default, we inherit from BasePlugin
+        # NB the choice of algorithm and executionOrder is arbitrary
+        singleFrame, forced = measBase.wrapSimpleAlgorithm(measBase.PsfFluxAlgorithm,
+                                                           Control=measBase.PsfFluxControl,
+                                                           executionOrder=0.0, doRegister=False)
+        self.assertEqual(singleFrame.getTransformClass(), measBase.BasePlugin.getTransformClass())
+        # Unless we override
+        singleFrame, forced = measBase.wrapSimpleAlgorithm(measBase.PsfFluxAlgorithm,
+                                                           Control=measBase.PsfFluxControl,
+                                                           executionOrder=0.0, doRegister=False,
+                                                           TransformClass=measBase.PassThroughTransform)
```

Comment:
Looks like there should be a check that singleFrame.getTransformClass() is PassThroughTransform here.